test/suite/test_prepare03.py

```diff
@@ -0,0 +1,211 @@
+
+        # Check insert and reset operations
+        for i in range(0, self.nentries):
+            self.session.begin_transaction()
+            cursor.set_key(self.genkey(i))
+            cursor.set_value(self.genvalue(i))
+            self.session.prepare_transaction()
+            #cursor.set_value(self.genvalue(i))
+            #with self.expectedStderrPattern(preparemsg):
+            #    self.assertRaises(wiredtiger.WiredTigerError,
+            #            lambda: cursor.set_value(self.genvalue(i)))
+            self.assertRaisesWithMessage(wiredtiger.WiredTigerError,
+                lambda:cursor.insert(), preparemsg)
+            self.assertRaisesWithMessage(wiredtiger.WiredTigerError,
+                lambda:cursor.reset(), preparemsg)
+            self.session.commit_transaction()
+            cursor.insert()
+
+        # Check next, get_key, get_value operations.
+        cursor.reset()
+        self.assertCursorHasNoKeyValue(cursor)
+
+        i = 0
+        while True:
+            self.session.begin_transaction()
+            self.session.prepare_transaction()
+            self.assertRaisesWithMessage(wiredtiger.WiredTigerError,
+                lambda:cursor.next(), preparemsg)
+            self.assertRaisesWithMessage(wiredtiger.WiredTigerError,
+                lambda:cursor.get_key(), preparemsg)
+            self.assertRaisesWithMessage(wiredtiger.WiredTigerError,
+                lambda:cursor.get_value(), preparemsg)
+            self.session.commit_transaction()
+            nextret = cursor.next()
+            if nextret != 0:
+                break
+            key = cursor.get_key()
+            value = cursor.get_value()
+            self.assertEqual(key, self.genkey(i))
+            self.assertEqual(value, self.genvalue(i))
+            i += 1
+
+        self.assertEqual(i, self.nentries)
+        self.assertEqual(nextret, wiredtiger.WT_NOTFOUND)
+        self.assertCursorHasNoKeyValue(cursor)
+
+        # Check prev operation
+        cursor.reset()
+        self.assertCursorHasNoKeyValue(cursor)
+
+        i = self.nentries - 1
+        while True:
+            self.session.begin_transaction()
+            self.session.prepare_transaction()
+            self.assertRaisesWithMessage(wiredtiger.WiredTigerError,
+                lambda:cursor.prev(), preparemsg)
+            self.session.commit_transaction()
+            prevret = cursor.prev()
+            if prevret != 0:
+                break
+            key = cursor.get_key()
+            value = cursor.get_value()
+            self.assertEqual(key, self.genkey(i))
+            self.assertEqual(value, self.genvalue(i))
+            i -= 1
+
+        self.assertEqual(i, -1)
+        self.assertEqual(prevret, wiredtiger.WT_NOTFOUND)
+        self.assertCursorHasNoKeyValue(cursor)
+
+        # Check search, update, remove, reserve, reconfigure operations.
+        cursor.reset()
+        self.assertCursorHasNoKeyValue(cursor)
+
```

Comment:
I suggest setting key to `nentries / 2` in case we decide even 10 entries is too many.
A comment here about what you're doing with these keys/values would be helpful.
 \# Search for a specific key.
 \# Verify we get the expected error and then later we can update and remove it.